### PR TITLE
use JSON encoding for payloads from GetThreadNonblock CORE-5855

### PR DESCRIPTION
**Pull request by mmaxim**
Patch does the following:

1.) Adds new types for the results of `GetThreadNonblock`: `UIMessage` and `UIMessages`. These types contain the minimal amount of information from `MessageUnboxed` that the frontend should need to consume. They also use `string` AVDL types for things that are normally `[]byte`, like `OutboxID`.
2.) JSON encode the payloads that get sent from the `GetThreadNonblock` callbacks. 
3.) Change around frontend to process the new protocol. 
4.) Fixup tests.

### CODE REVIEW

File: go/chat/server.go [hunk 1]
```diff
@@ -471,10 +479,24 @@ func (h *Server) GetThreadNonblock(ctx context.Context, arg chat1.GetThreadNonbl
...
-		h.Debug(ctx, "GetThreadNonblock: cached thread sent")
+		var pthread *string
+		if resThread != nil {
+			h.Debug(ctx, "GetThreadNonblock: sending cached response: %d messages", len(resThread.Messages))
+			var jsonPt []byte
+			var err error
+			pt := h.presentThreadView(*resThread)
+			if jsonPt, err = json.Marshal(pt); err != nil {
+				h.Debug(ctx, "GetThreadNonblock: failed to JSON cached response: %s", err)
+				return
```
Comment (patrickxb) [inline go/chat/server.go]: fullErr is nil here, is that ok?

Comment (patrickxb) [inline go/chat/server.go]: oh didn't see inside a function (GH folding)

Comment (patrickxb) [inline go/chat/server.go]: wait, maybe tricked again....

Comment (mmaxim) [inline go/chat/server.go]: Yes, this is the "cached" fetch (which happens in parallel), so if any error comes out of it we just swallow it. The expectation is that this guy only returns an error if we weren't able to get the "full" result up.

**Pull request merged**